File: nucleus.py
n = 4
arr = [[1,2,3,5],[4,5,6,7],[7,8,9,8],[2,3,4,7]]
l =[]

print(arr)

for i in range(n):
	m=i
	j=0
	s=0
	while(j<=i and m>=0):
		s += arr[m][j]
		
		if m==0:
			break
		else:
			m=m-1
		j=j+1

	print(s)

	l.append(s)
	continue;

i = n-1
for j in range(1,n):
	m = i
	p = j

	s=0
	while(p<=i):
		s += arr[m][p]

		m = m-1
		p = p+1

	print(s)
	l.append(s)
	continue

File: nucleus.cpp
#include<bits/stdc++.h>
#include<iostream> 
using namespace std;

int main()
{
	int n;
	cin>>n;
	int arr[n][n];
	for(int i=0; i<n;i++){
		for(int j=0; j<n;j++){
		cin>> arr[i][j];
	}
	}

	for(int i=0; i<n;i++){
		for(int j=0; j<n;j++){
		cout<< arr[i][j]<<" ";
	}
	}

	int s[7] = {};
	
	int sum=0, j=0;

	for(int i=0, j=0;i<n;i++){
		if(i==0 && j==0){ sum+=arr[i][j]; s[i]=sum; continue;}
		int m=i;
		
			while(j<n && m>=0){
			sum += arr[m][j];
			m=m-1;
			j=j+1;
		}
		s[i] = sum;
		sum=0;

	}

	int i=n;
	for(int j=0; j<n; j++)
	{
		int m=i;
		int l=m+j;
		
			while(j<n && m>=0){
			sum += arr[m][j];
			m=m-1;
			j=j+1;
		}
		s[l] = sum;
		sum=0;
		
	}

	for(int k=0; k< 7; k++){
		cout<<" Output : " << s[k] << " ";
	}
	
}
